pkg/network.go

```diff
@@ -47,7 +48,7 @@ type Network struct {
 	//unique id inside the reservation is an autoincrement (USE AS NET_ID)
 	NetID NetID `json:"net_id"`
 	// IP range of the network, must be an IPv4 /16
-	IPRange *net.IPNet `json:"ip_range"`
+	IPRange schema.IPRange `json:"ip_range"`
```

Comment:
schema.IPRange is not a (jsx) schema type and not generated. It's just a wrapper around net.IPNet that can do json serialization/deserialization. Then i use this type in generated schema structures for the same reason. 

Because i switched to this type, i was able to remove lots of custom Marshaling code.

Reply:
Ok but I still don't like we depend on a type that is versioning in another module. I get that it indeed removed quite some code so that's good. But let's then move this type into the network package. Also, is IPRange the proper name for it ?

Reply:
the name unfortunately (while not the best) is inherited from the jsx `iprange` type. About moving the type out of the schema module is not okay, because a generated type (like the generated types in gedis client) will reference the `schema.IPRange` type for each `iprange` type in jsx schema. 

Copying the code also is not a good solution, but if you really insist, i can create a type alias in the network package, so if we want to separate the types later, we can still do it. What do u think?

Reply:
> the name unfortunately (while not the best) is inherited from the jsx iprange type.

I know but we said we don't want to leak anything from the JSX types into the main code base. 

>  About moving the type out of the schema module is not okay, because a generated type (like the generated types in gedis client) will reference the schema.IPRange type for each iprange type in jsx schema.

I don't understand what is the problem ? A conversion between types is already happening in gedis package anyhow.  And if anything, types from our code base should spreads into the client that implement interface with outside components, not the other way around where we end up using stuff from outside into the core core of 0-OS.

> Copying the code also is not a good solution, but if you really insist, i can create a type alias in the network package, so if we want to separate the types later, we can still do it. What do u think?

Not happy with that either. If you want to have a type that implement the marshaling around an `net.IPNet`, fine, but then it must be define in our network package. Not an alias from another type and not a type from an external package.